scheme/common/utils.go

```diff
@@ -35,3 +39,26 @@ func GetMandatoryPathSegment(key string, fields map[string]*structpb.Value) (str
 
 	return segment, nil
 }
+
+func GetPublicKeyFromTa(ta []byte) (crypto.PublicKey, error) {
+	block, rest := pem.Decode(ta)
+
+	if block == nil {
+		log.Error("Could not get TA PEM Block ValidateEvidenceIntegrity")
+		return nil, errors.New("could not extract trust anchor PEM block")
+	}
+
+	if len(rest) != 0 {
+		return nil, errors.New("trailing data found after PEM block")
+	}
+
+	if block.Type != "PUBLIC KEY" {
+		return nil, fmt.Errorf("unsupported key type: %q", block.Type)
+	}
+
+	pk, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse public key: %w", err)
+	}
+	return pk, nil
```

Comment:
After checking x.509 error return parameters for ECDSA, I think it does add a value as otherwise the errors returned will be like below:
x509: invalid ecdsa parameter
x509: failed to unmarshal elliptic curve point

So with the added information it sets the top level context with the specific reason as to why parse failed.